run_tribler.py

```diff
@@ -1,30 +1,75 @@
 
 from check_os import check_environment
 
+
+def setup_logging():
+    """
+    Setup logging to write logs to files inside \
+    .Tribler directory in each platforms
+    """
+    # First check if logger.conf is present or not
+    if getattr(sys, 'frozen', False):
+        # If the application is run as a bundle, the pyInstaller bootloader
+        # extends the sys module by a flag frozen=True and sets the app
+        # path into variable _MEIPASS'.
+        application_path = os.environ.get("_MEIPASS", os.path.abspath("."))
+    else:
+        application_path = os.path.dirname(os.path.abspath(__file__))
+
+    log_config = os.path.join(application_path, "logger.conf")
+
+    if not os.path.exists(log_config):
+        print "Log configuration file not found at location '%s'" % log_config
+        return
+
+    if sys.platform == 'linux2' or sys.platform == 'darwin':
+        log_directory = os.path.expanduser('~/.Tribler/logs')
+    elif sys.platform == 'win32':
+        log_directory = os.path.join(os.getenv('APPDATA'), '.Tribler\\logs')
+    else:
+        logging.error('Logging to file is not enabled')
+        return
+
+    if not os.path.exists(log_directory):
+        os.makedirs(log_directory)
+
+    logging.info_log_file = '%s/tribler-info.log' % log_directory
+    logging.error_log_file = '%s/tribler-error.log' % log_directory
+    logging.config.fileConfig(log_config, disable_existing_loggers=False)
+
+
 if __name__ == "__main__":
     # Exit if we cant read/write files, etc.
     check_environment()
 
     multiprocessing.freeze_support()
 
-    from TriblerGUI.tribler_app import TriblerApplication
-    from TriblerGUI.tribler_window import TriblerWindow
+    # Set up logging
+    setup_logging()
+
+    try:
+        from TriblerGUI.tribler_app import TriblerApplication
+        from TriblerGUI.tribler_window import TriblerWindow
 
-    app = TriblerApplication("triblerapp", sys.argv)
+        app = TriblerApplication("triblerapp", sys.argv)
 
-    if app.is_running():
-        for arg in sys.argv[1:]:
-            if os.path.exists(arg):
-                app.send_message("file:%s" % arg)
-            elif arg.startswith('magnet'):
-                app.send_message(arg)
-        sys.exit(1)
+        if app.is_running():
+            for arg in sys.argv[1:]:
+                if os.path.exists(arg):
+                    app.send_message("file:%s" % arg)
+                elif arg.startswith('magnet'):
+                    app.send_message(arg)
+            sys.exit(1)
 
-    window = TriblerWindow()
-    window.setWindowTitle("Tribler")
-    app.set_activation_window(window)
-    app.parse_sys_args(sys.argv)
-    sys.exit(app.exec_())
+        window = TriblerWindow()
+        window.setWindowTitle("Tribler")
+        app.set_activation_window(window)
+        app.parse_sys_args(sys.argv)
+        sys.exit(app.exec_())
+    except ImportError as ie:
+        print "Import error: {0}".format(ie)
+        logging.exception(ie)
```

Comment:
Also, related to #3120, could you call `check_os.error_and_exit()` here with the import message.

Reply:
yes, that would be nice